--- piglit/CMakeLists.txt	2011-08-10 08:25:17.000000000 -0700
+++ piglit/CMakeLists.txt	2011-08-11 21:02:18.434579000 -0700
@@ -77,11 +77,14 @@ else (APPLE)
 	)
 endif (APPLE)
 
+find_library(OPENGL_gles2_LIBRARY NAMES GLESv2)
+
+if(OPENGL_gles2_LIBRARY)
 FIND_LIBRARY(OPENGL_egl_LIBRARY
       NAMES EGL
       PATHS /usr/lib
 )
-find_library(OPENGL_gles2_LIBRARY NAMES GLESv2)
+endif(OPENGL_gles2_LIBRARY)
 
 # Put all executables into the bin subdirectory
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${piglit_BINARY_DIR}/bin)
