diff -urN ltp-full-20091231_vanilla/runtest/syscalls ltp-full-20091231/runtest/syscalls
--- ltp-full-20091231_vanilla/runtest/syscalls	2010-01-05 17:16:06.000000000 -0800
+++ ltp-full-20091231/runtest/syscalls	2010-01-05 17:30:32.000000000 -0800
@@ -1245,7 +1245,7 @@
 vhangup02 vhangup02
 
 #vmsplice test cases
-vmsplice01 vmsplice01
+#vmsplice01 vmsplice01
 
 wait02 wait02
 
--- ltp-full-20091231_vanilla/testcases/kernel/syscalls/epoll/epoll-ltp.c	2010-03-22 17:25:18.000000000 -0700
+++ ltp-full-20091231/testcases/kernel/syscalls/epoll/epoll-ltp.c	2010-03-22 17:27:06.000000000 -0700
@@ -450,7 +450,7 @@
 			ev_ptr = &event;
 			break;
 		case 1:	/* Pass unaligned struct */
-			unaligned_event_ptr->data.u64 = 0;
+			memset(&unaligned_event_ptr->data.u64, 0, sizeof(unaligned_event_ptr->data.u64));
 			ev_ptr = unaligned_event_ptr;
 			break;
 		case 2:
