diff -Naur misc.c misc.c
--- misc.c	2008-08-07 09:50:29.000000000 -0700
+++ misc.c	2010-04-05 12:06:14.000000000 -0700
@@ -96,8 +96,11 @@
 rdtsc(int argc, const char *argv[], const struct cmd_info *info)
 {
 	unsigned long long tsc;
-
+#ifdef __arm__
+	tsc = 0;
+#else
 	rdtscll(tsc);
+#endif
 	printf("0x%016llx\n", tsc);
 
 	return 0;
@@ -110,14 +113,16 @@
 	if (set_cpu_affinity(cpu) < 0) {
 		return -1;
 	}
-
+#ifdef __arm__
+	return -1;
+#else
 	asm volatile (
 	      "cpuid\n\t"
 	      : "=a" (data[0]), "=b" (data[1]), "=c" (data[2]), "=d" (data[3])
 	      : "0" (function), "2" (index)
 	      : "memory"
 	);
-
+#endif
 	return 0;
 }
 
