--- configure.old	2011-07-01 15:04:04.745547173 -0700
+++ configure	2011-07-01 15:11:37.635573853 -0700
@@ -13411,33 +13411,7 @@
 fi
 
 # Check for libgsm
-found_libgsm=yes
-for ac_header in gsm/gsm.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "gsm/gsm.h" "ac_cv_header_gsm_gsm_h" "$ac_includes_default"
-if test "x$ac_cv_header_gsm_gsm_h" = x""yes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_GSM_GSM_H 1
-_ACEOF
-
-else
-  for ac_header in gsm.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "gsm.h" "ac_cv_header_gsm_h" "$ac_includes_default"
-if test "x$ac_cv_header_gsm_h" = x""yes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_GSM_H 1
-_ACEOF
-
-else
-  found_libgsm=no
-fi
-
-done
-
-fi
-
-done
+found_libgsm=no
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gsm_create in -lgsm" >&5
 $as_echo_n "checking for gsm_create in -lgsm... " >&6; }
@@ -13445,7 +13419,6 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lgsm  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13476,18 +13449,11 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_gsm_gsm_create" >&5
 $as_echo "$ac_cv_lib_gsm_gsm_create" >&6; }
 if test "x$ac_cv_lib_gsm_gsm_create" = x""yes; then :
-  GSM_LIBS="$GSM_LIBS -lgsm"
 else
   found_libgsm=no
 fi
 
-if test "$found_libgsm" = yes; then
-
-$as_echo "#define EXTERNAL_GSM 1" >>confdefs.h
-
-else
-    LIBGSM_LIBADD=../libgsm/libgsm.la
-fi
+  LIBGSM_LIBADD=../libgsm/libgsm.la
  if test x$found_libgsm = xyes; then
   EXTERNAL_GSM_TRUE=
   EXTERNAL_GSM_FALSE='#'
